src/utils/convertTimeStringToObj.ts

```diff
@@ -0,0 +1,36 @@
+import { TimeObjectSchema } from 'serina.schema';
+import { POST_MERIDIEM, ALL_MERIDIEM } from 'filters/time/time.constants';
+import contains from './contains';
+import { remove, isValidTime } from 'utils';
+
+function convertTimeStringToObj(time: string): TimeObjectSchema {
+    let hour;
+    let minute;
+
+    if (contains(time, `( )?${ALL_MERIDIEM}`, false)) {
+        // 12H
+        const timeWithMeridiem = remove(time, `( )?${ALL_MERIDIEM}`, false);
+        const timeSplit = timeWithMeridiem.split(':');
+        hour = parseInt(timeSplit[0], 10);
+        minute = parseInt(timeSplit[1], 10);
+
+        if (isNaN(minute)) minute = 0;
```

Comment:
I think if we are going to allow whole hours to be written as "8" rather than "8:00" the above logic might change slightly. The split on (':') is still fine, but then we need to check for minute to be undefined.

Reply:
See without doing that, this logic is simpler too lol